cli/spec/spec_helper.rb

```diff
@@ -46,25 +46,18 @@
 
   config.around(:each) do |example|
     catch :exit_with_error do
+      $stdout = $stderr = StringIO.new
+
       begin
         example.run
-      rescue SystemExit
-        puts "Got SystemExit: #{$!.message} - Exit code: #{$!.status}"
+      rescue SystemExit => exc
+        fail "SystemExit with code #{exc.status}: \n#{$stderr.string}"
```

Comment:
This needs something to include the output from `exit_with_error`, or specs that fail with `SystemExit` are impossible to figure out, because the actual `exit` call is so obfuscated: e.g. this case is failing in [`Kontena::Cli::Common#confirm`](https://github.com/kontena/kontena/blob/d2a8ef67cffbe62805f0b61419f9a74df84c3ae8/cli/lib/kontena/cli/common.rb#L270) because of the lack of `--force`, but it's next to impossible to figure that out:

```
  3) Kontena::Cli::Stacks::UpgradeCommand#execute --no-deploy option does not trigger deploy
     Failure/Error: fail "SystemExit with code #{exc.status}: #{exc.message}"
     
     RuntimeError:
       SystemExit with code 1: exit
     # ./spec/spec_helper.rb:52:in `rescue in block (3 levels) in <top (required)>'
     # ./spec/spec_helper.rb:49:in `block (3 levels) in <top (required)>'
     # ./spec/spec_helper.rb:48:in `catch'
     # ./spec/spec_helper.rb:48:in `block (2 levels) in <top (required)>'
     # ------------------
     # --- Caused by: ---
     # SystemExit:
     #   exit
     #   ./lib/kontena/cli/common.rb:168:in `exit'
```

Running the specs with `DEBUG=true` to not capture `$stdout`/`$stderr` does not help at all, because it also changes the behavior of `confirm`, and also the error handling for exceptions within the command themselves...